lib/src/install/config.rs

```diff
@@ -171,9 +192,18 @@ pub(crate) fn load_config() -> Result<Option<InstallConfiguration>> {
         if let Some(config) = config.as_mut() {
             if let Some(install) = c.install {
                 tracing::debug!("Merging install config: {install:?}");
-                config.merge(install);
+                config.merge(install, &env);
             }
         } else {
+            // If arch is specified and doesn't match the current arch, remove kargs
```

Comment:
I think the goal of this PR was just to do the passing of the environment without functional change, but this still does the functional change of handling arch-specific install args right?

So the "prep change" here would also drop this and the tests and really everything related to actually putting content in that env function.

Alternatively, I'm also fine if we just roll together the new API alongside the prep work!  IOW you could also just rework the commit message to say something like:

```
install: Add support for architecture filtering for kargs

The use case here is that some kernel arguments are
architecture specific, such as the `console=` kargs.
This allows a generic base image to default to including
arch-specific values post install.
```